pkg/solana/txm/pendingtx.go

```diff
@@ -271,6 +277,30 @@ func (c *pendingTxContext) OnProcessed(sig solana.Signature) (string, error) {
 	})
 }
 
+// IsBlockhashExpired returns whether or not the blockhash of a transaction has expired when compared to currHeight.
+func (c *pendingTxContext) IsBlockhashExpired(id string, currHeight uint64) bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	tx, exists := c.broadcastedTxs[id]
+	if !exists {
+		return false
+	}
+	return tx.lastValidBlockHeight < currHeight
+}
+
+// UpdateBlockhash updates the blockhash and last valid height for a transaction when expired and needs bumping.
+func (c *pendingTxContext) UpdateBlockhash(id string, blockhashResult *rpc.GetLatestBlockhashResult) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if tx, exists := c.broadcastedTxs[id]; exists {
+		tx.tx.Message.RecentBlockhash = blockhashResult.Value.Blockhash
```

Comment:
I think this is ok since this is essentially just the base tx. We should also clear the `msg.signatures` though since those correlate to the expires transaction which we don't need to track anymore for confirmation